### ISSUE TITLE
Exception when deleting selection that spans the last item in a list and paragraph below

### ISSUE THREAD

**Issue by jjallaire** [ProseMirror/prosemirror#1010]
### Issue details

When deleting the last item of a list along with a paragraph immediately below, a JavaScript exception (`TypeError: Cannot read property 'content'`) occurs:

```
M13 prosemirror.js:1 Uncaught TypeError: Cannot read property 'content' of undefined
    at Zo.end (VM13 prosemirror.js:1)
    at e.Bi.deleteRange (VM13 prosemirror.js:1)
    at e.Bi.replaceRange (VM13 prosemirror.js:1)
    at e.Ui.replace (VM13 prosemirror.js:1)
    at e.replace (VM13 prosemirror.js:1)
    at e.deleteSelection (VM13 prosemirror.js:1)
    at Array.Sr (VM13 prosemirror.js:1)
    at VM13 prosemirror.js:1
    at us.<anonymous> (VM13 prosemirror.js:1)
    at VM13 prosemirror.js:1
prosemirror.js:1 Uncaught TypeError: Cannot read property 'content' of undefined
    at Zo.end (prosemirror.js:1)
    at e.Bi.deleteRange (prosemirror.js:1)
    at e.Bi.replaceRange (prosemirror.js:1)
    at e.Ui.replace (prosemirror.js:1)
    at e.replace (prosemirror.js:1)
    at e.deleteSelection (prosemirror.js:1)
    at Array.Sr (prosemirror.js:1)
    at prosemirror.js:1
    at us.<anonymous> (prosemirror.js:1)
    at prosemirror.js:1
```

For example, this content would exhibit the issue:

```html
<ul>
  <li>Item 1</li>
  <li>Item 2</li>
</ul>

<p>Try selecting this text along with Item 2 above, when you press
backspace to delete the selection, you will see this JS exception</p>
```

### Steps to reproduce

I forked the prosemirror-demo-basic to include base content that exhibits the issue here:

https://glitch.com/edit/#!/stealth-individual?path=index.html

### ProseMirror version

Version loaded from https://prosemirror.net/examples/prosemirror.js

### Affected platforms

- [x] Chrome (v78.0.3904)
- [x] Firefox (v70.0.1)
- [x] Safari (v13.0.3)
- [ ] Internet Explorer

### Screenshots / Screencast (Optional)

Screencast: https://recordit.co/jP07uZfsrq

**Comment by marijnh** [comment]
Thanks for the report. Attached patch (released as prosemirror-transform 1.2.3) should help.

**Comment by jjallaire** [comment]
Thanks for the patch! Glad it was a straightforward fix.
